src/python/pants/help/help_info_extracter.py

```diff
@@ -781,6 +801,86 @@ def load() -> PluginAPITypeInfo:
             }
         )
 
+    @classmethod
+    def get_backend_help_info(cls, options: Options) -> LazyFrozenDict[str, BackendHelpInfo]:
+        DiscoveredBackend = namedtuple(
+            "DiscoveredBackend", "provider, name, register_py, enabled", defaults=(False,)
+        )
+
+        def discover_source_backends(root: Path, is_source_root: bool) -> set[DiscoveredBackend]:
+            provider = root.name
+            source_root = f"{root}/" if is_source_root else f"{root.parent}/"
+            register_pys = root.glob("**/register.py")
+            backends = {
+                DiscoveredBackend(
+                    provider,
+                    str(register_py.parent).replace(source_root, "").replace("/", "."),
+                    str(register_py),
+                )
+                for register_py in register_pys
+            }
+            return backends
+
+        def discover_plugin_backends(entry_point_name: str) -> set[DiscoveredBackend]:
+            backends = {
+                DiscoveredBackend(
+                    entry_point.dist.project_name,
+                    entry_point.module_name,
+                    str(
+                        Path(entry_point.dist.location)
+                        / (entry_point.module_name.replace(".", "/") + ".py")
+                    ),
+                    True,
+                )
+                for entry_point in pkg_resources.iter_entry_points(entry_point_name)
+                if entry_point.dist is not None
+            }
+            return backends
+
+        global_options = options.for_global_scope()
+        builtin_backends = discover_source_backends(
+            Path(pants.backend.__file__).parent.parent, is_source_root=False
```

Comment:
Do you know if this works when using `./pants_from_sources`?

Reply:
Confirmed, it works:
```
╰─❯ ./pants_from_sources backends --help
...
[ ] examples.bash                                                                                                                                                                                                                  [pants-plugins]
     The entry point for our `examples.bash` plugin.

     See https://www.pantsbuild.org/v2.0/docs/plugins-overview.
[ ] examples.bash.lint.shellcheck                                                                                                                                                                                                  [pants-plugins]
     The entry point for our `examples.bash.lint.shellcheck` plugin.

     See https://www.pantsbuild.org/v2.0/docs/plugins-overview.

     Note that we use a separate `register.py` than the main `examples/bash/register.py` so that users must opt into activating Shellcheck. We could have instead registered the Shellcheck rules there if we were okay with Shellcheck always
     being activated when the Bash backend is activated.
[ ] examples.bash.lint.shfmt                                                                                                                                                                                                       [pants-plugins]
     The entry point for our `examples.bash.lint.shfmt` plugin.

     See https://www.pantsbuild.org/v2.0/docs/plugins-overview.

     Note that we use a separate `register.py` than the main `examples/bash/register.py` so that users must opt into activating shfmt. We could have instead registered the shfmt rules there if we were okay with shfmt being activated when
     the Bash backend is activated.
```